styles/main/_form.scss

```diff
@@ -0,0 +1,373 @@
+	}
+
+	&:checked::before {
+		// Use `visibility` instead of `display` to avoid layout recalculation
+		visibility: visible;
+	}
+}
+
+input[type="checkbox"].slider {
+	top: 5px;
+	height: 22px;
+	width: 42px;
+	border: 1px solid rgba(0, 0, 0, 0.2);
+	box-shadow: 0 1px 0 rgba(255, 255, 255, 0.02);
+	border-radius: 11px;
+	background: $colorFormElementBg;
+
+	&::before {
+		content: "";
+		background-color: $colorFormElementAccent;
+		height: 14px;
+		width: 14px;
+		left: 3px;
+		top: 3px;
+		border: none;
+		border-radius: 7px;
+		visibility: visible;
+	}
+
+	&:checked {
+		background-color: $colorFormElementAccent;
+	}
+
+	&:checked::before {
+		left: auto;
+		right: 3px;
+		background-color: $colorFormElementFg;
+	}
+}
+
+div.select {
+	font-size: 12px;
+	background: $colorFormElementBg;
+	border-radius: 3px;
+	border: 1px solid black(0.2);
+	box-shadow: 0 1px 0 white(0.02);
+
+	&::after {
+		position: absolute;
+		content: "≡";
+		right: 8px; // must match the right padding of the `<select>` element
+		top: 3px; // must match the top padding of the `<select>` element
+		color: $colorFormElementAccent;
+		font-size: 16px;
+		font-weight: bold;
+		pointer-events: none;
+	}
+}
+
+select {
+	padding-left: 8px;
+	padding-right: 8px; // must match the right position of the `div.select::after`
+	appearance: none;
+	background: none;
+}
+
+select option {
+	padding: 2px 0;
+	transition: none;
+}
+
+// 2. Input groups
+//
+// Only <div> are expected as direct children of a form.
+// These <div> constitute a so-called "input group".
+// An input group consists of the following elements in the prescribed
+// order:
+//
+//  - an optional `<label>`
+//  - exactly one mandatory HTML form element:
+//      -- `<input>`,
+//      -- `<div class='select`> nesting a `<select>`,
+//      -- `<textarea>`, or
+//      -- `<output>`
+//  - zero or more optional `<p>` elements with a descriptive text
+//  - zero or more optional `<div>` with nested input groups
+//
+// One note about `<select>`: This element is hard to style with CSS,
+// because it is treated as a replaced element. Hence, `<select>` needs to
+// be wrapped into an additional `<div>` and most styles are applied to
+// this `<div>`.
+// See: https://developer.mozilla.org/en-US/docs/Learn/Forms/Advanced_form_styling
+//
+// An input group can have one out of four layouts which determines
+// how the label, the form element, the description and the nested
+// input groups are arranged:
+//
+//  1. `stacked`
+//      All elements are vertically stacked above each other.
+//
+//  2. `compact`
+//      The label and the form element are on the same horizontal line.
+//      The other elements follow below with a hanging indention and
+//      are aligned to the form element
+//
+//  3. `compact-inverse`
+//      The form element and the label are on the same horizontal line.
+//      The other elements follow below with a hanging indention and are
+//      aligned to the form element
```

Comment:
The explanation for (2) and (3) is currently the same, which is unhelpful.

Also, I'm a bit surprised by the word _indention_. While apparently an existing word, I think I've always seen _indentation_ used in this context?

Reply:
I think "indentation" is more correct. "Indent" is more common because that's what Word uses. Also, e.g. `text-indent` in CSS.

Reply:
> The explanation for (2) and (3) is currently the same, which is unhelpful.

Actually not :laughing: 

> 2. The **label** and the **form element** are on the same horizontal line.

vs.

> 3. The **form element** and the **label** are on the same horizontal line.

The order of words was swapped on purpose. But I agree that this was extremely subtle. :laughing: I made it more explicit in https://github.com/LycheeOrg/Lychee-front/pull/322/commits/4320beb8e5c2347321aa019d4561b012de73d9d7.

> Also, I'm a bit surprised by the word indention. While apparently an existing word, I think I've always seen indentation used in this context?

Also fixed in https://github.com/LycheeOrg/Lychee-front/pull/322/commits/4320beb8e5c2347321aa019d4561b012de73d9d7.